recipes/gmp/all/conanfile.py

```diff
@@ -65,106 +61,105 @@ def config_options(self):
             "--enable-fat={}".format(yes_no(self.options.get_safe("enable_fat", False))),
             "--enable-cxx={}".format(yes_no(self.options.enable_cxx)),
-            "--enable-shared={}".format(yes_no(self.options.shared)),
-            "--enable-static={}".format(yes_no(not self.options.shared)),
-            "--srcdir={}".format(os.path.join(self.source_folder, self._source_subfolder).replace("\\", "/")),
-        ]
-        if self._is_msvc:
-            configure_args.extend([
+            "--srcdir={}".format(self.source_folder.replace("\\", "/")),
+        ])
+        if is_msvc(self):
+            tc.configure_args.extend([
                 "ac_cv_c_restrict=restrict",
                 "gmp_cv_asm_label_suffix=:",
                 "lt_cv_sys_global_symbol_pipe=cat",  # added to get further in shared MSVC build, but it gets stuck later
             ])
-            if not (self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version) < 12):
-                autotools.flags.append("-FS")
-            autotools.cxx_flags.append("-EHsc")
-        autotools.configure(args=configure_args, configure_dir=self._source_subfolder)
-        return autotools
-
-    @contextlib.contextmanager
-    def _build_context(self):
-        if self._is_msvc:
-            with tools.vcvars(self):
-                yasm_machine = {
-                    "x86": "x86",
-                    "x86_64": "amd64",
-                }[str(self.settings.arch)]
-                env = {
-                    "CC": "cl -nologo",
-                    "CCAS": "{} -a x86 -m {} -p gas -r raw -f win32 -g null -X gnu".format(os.path.join(self.build_folder, "yasm_wrapper.sh").replace("\\", "/"), yasm_machine),
-                    "CXX": "cl -nologo",
-                    "AR": "{} lib".format(self._user_info_build["automake"].ar_lib.replace("\\", "/")),
-                    "LD": "link -nologo",
-                    "NM": "python {}".format(tools.unix_path(os.path.join(self.build_folder, "dumpbin_nm.py"))),
-                }
-                with tools.environment_append(env):
-                    yield
-        else:
-            yield
+            tc.extra_cxxflags.append("-EHsc")
+            if (str(self.settings.compiler) == "Visual Studio" and Version(self.settings.compiler.version) >= "12") or \
+               (str(self.settings.compiler) == "msvc" and Version(self.settings.compiler.version) >= "180"):
+                tc.extra_cflags.append("-FS")
+                tc.extra_cxxflags.append("-FS")
+        env = tc.environment()
+        if is_msvc(self):
+            yasm_wrapper = unix_path(self, os.path.join(self.source_folder, "yasm_wrapper.sh"))
+            yasm_machine = {
+                "x86": "x86",
+                "x86_64": "amd64",
+            }[str(self.settings.arch)]
+            ar_wrapper = unix_path(self, self._user_info_build["automake"].ar_lib)
```

Comment:
I was under the impression msys had the AR tool with 🤔 looking at my git bash prompt it does?

Reply:
What do you mean by AR tool? ar-lib in automake recipe is not the GNU archiver, but ar-lib.sh, a wrapper for Visual Studio in order to translate GNU AR options to lib archiver options of Visual Studio.